spring-cloud-starter-stream-source-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/source/SftpSourceConfiguration.java

```diff
@@ -208,4 +288,107 @@ public Message transformSftpMessage(Message message) {
 				.setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.TEXT_PLAIN).build();
 	}
 
+	@Bean
+	@ConditionalOnProperty("sftp.multi-source")
+	public ListFilesRotator rotator(SftpSourceProperties properties, DelegatingFactoryWrapper factory) {
+		return new ListFilesRotator(properties, factory);
+	}
+
+}
+
+class ListFilesRotator extends AbstractMessageSourceAdvice {
```

Comment:
Why this class is not an inner one?
May be it should go to the top level even if we continue to keep it as package protected.
Otherwise, please, move it to the real inner class.

Reply:
Why is this important?

Reply:
Well, that's so unusual... You don't  do such a thing "several top-level classes in the same file" in other places...
That's hard to adjust to different code base in different project from the same person...
Here is some explanation though: https://stackoverflow.com/questions/360643/is-it-a-bad-practice-to-have-multiple-classes-in-the-same-file

🤷‍♂️